src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
          List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
          Printf.printf "About to fixup the fd\n%!";
-         ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+         Lwt.catch
```

Comment:
It should be fine by design of the code above it, as before most of the calls are safe and most of the ones that are not are protected. This said, there is a chance of leak if we fail in closing the `fd_sock2` (that should not happen unless other major issues). I think this is minor compared to the risks that we had before.

Just stressing my comment above. We should really consider rewriting this as a proper service using tested upstream libraries and a cleaner design, but this is unfortunately out of the scope of this PR.